DuckDuckGo/AnimatableTypingText.swift

```diff
@@ -0,0 +1,153 @@
+//
+//  AnimatableTypingText.swift
+//  DuckDuckGo
+//
+//  Copyright © 2024 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import SwiftUI
+import Core
+import Combine
+
```

Comment:
I found the code for animating the text in macOS -> [DaxSpeech](https://github.com/duckduckgo/macos-browser/blob/main/DuckDuckGo/Onboarding/View/DaxSpeech.swift). It was slightly coupled with the OnboardingModel so I’ve extracted to be more a generic view.

I had to make a change as the macOS implementation uses [autoconnect()](https://duckduckgo.com/?q=apple+documentation+autoconnect&atb=v413-1-wb&ia=web) and fire events immediately. 

On Android I noticed that in some scenarios we have to start animating the copy when another animation finishes. I moved the logic in the view viewmodel and wrote tests.